src/main/kotlin/com/sourcegraph/cody/auth/SelectOneOfTheAccountsAsActive.kt

```diff
@@ -1,17 +1,29 @@
 package com.sourcegraph.cody.auth
 
+import com.intellij.openapi.diagnostic.Logger
 import com.intellij.openapi.project.Project
 import com.sourcegraph.cody.CodyToolWindowContent
+import com.sourcegraph.cody.config.CodyAccount
 import com.sourcegraph.cody.config.CodyAuthenticationManager
 import com.sourcegraph.cody.config.getFirstAccountOrNull
 import com.sourcegraph.cody.initialization.Activity
 
 class SelectOneOfTheAccountsAsActive : Activity {
+
+  private val logger = Logger.getInstance(SelectOneOfTheAccountsAsActive::class.java)
+
   override fun runActivity(project: Project) {
-    val codyAuthenticationManager = CodyAuthenticationManager.instance
-    if (codyAuthenticationManager.getActiveAccount(project) == null) {
-      val newActiveAccount = codyAuthenticationManager.getAccounts().getFirstAccountOrNull()
-      codyAuthenticationManager.setActiveAccount(project, newActiveAccount)
+    val authManager = CodyAuthenticationManager.instance
+    val token = System.getenv("SOURCEGRAPH_TEST_TOKEN")
+    if (token != null) {
+      val testAccount = CodyAccount("test", "Test User")
+      authManager.updateAccountToken(testAccount, token)
+      authManager.setActiveAccount(project, testAccount)
+      logger.warn("Test account with external SOURCEGRAPH_TEST_TOKEN selected as active")
+    }
+    if (authManager.getActiveAccount(project) == null) {
```

Comment:
huh! so you cannot have both test account and "regular" account! I assume this is intended

Reply:
Correct. I believe that if someone explicitly sets external token, ignoring that token and switching to a real account would be undesirable.